### ISSUE TITLE
Collect, preprocess data, build and train model 

### ISSUE THREAD

**Issue by DinhNhatKy** [dtvn-training/2024-HN-KYDN#6]
- [x] Research and select model and dataset
- [x] Preprocess data, crop data, generate triplet data, create dataloader
- [x] Build model, define loss function, train model
- [x] Evaluate and select model
- [x] Build distance calculate function (l2 and cosine) + anti-spoof model
- [x] Build interface
- [x] Infer and build pipeline
- [x] Test in many data gallery
- [x] Create simple log file

- Documentation: [Facenet](https://arxiv.org/abs/1503.03832)
Val dataset: [Lfw](https://vis-www.cs.umass.edu/lfw/)
Train dataset: [VNceleb](https://www.flickr.com/photos/148530340@N06/47824829632/in/dateposted-public/) and custom data
Model detection: [Mtcnn](https://kpzhang93.github.io/MTCNN_face_detection_alignment/index.html) and [yolo](https://docs.ultralytics.com/tasks/detect/#models)
Model recognition: Inceptionresnet use pretrained
Anti spoofing model (use pretrain model): [from github](https://github.com/OeslleLucena/FASNet)

- pipeline of project:
![pipeline](https://github.com/user-attachments/assets/418edef7-9001-4ad6-a8bf-546a59f26a3c)
